scripts/support/runner/nrfjprog.py

```diff
@@ -40,8 +40,13 @@ def get_board_snr_from_user(self):
         snrs = self.check_output(['nrfjprog', '--ids'])
         snrs = snrs.decode(sys.getdefaultencoding()).strip().splitlines()
 
-        if len(snrs) == 1:
-            return snrs[0]
+        if len(snrs) == 0:
+            raise Exception('"nrfjprog --ids" did not find a board; Is the board connected?')
```

Comment:
do we need to raise an exception? Can't we instead return something != 0 and let make/ninja output the error? the stack trace output is verbose and distracts the user from the actual useful output.

Reply:
It's more Pythonic to raise an exception here. I will figure out why the top-level zephyr_flash_debug.py isn't swallowing it and just printing the message.

Reply:
Here is a PR which swallows the stack trace and just prints an error message, unless verbose output was requested: https://github.com/zephyrproject-rtos/zephyr/pull/5434